Updated Distributed Network manual

fireclawthefox opened the pull request:
Added documentation for most of the Distributed Network parts

fireclawthefox committed:
Updated Distributed Network manual

Added documentation for most of the Distributed Network parts

fireclawthefox committed:
Fix linting errors

escaped trailing underscores in text

rdb commented on programming/networking/distributed.rst:
```diff
@@ -1,16 +0,0 @@
-that object. Updates to the object will automatically propagate to the copies.
-
-The distributed network is composed of several layers: the dc file which
-defines the communication, ServerRepositories which handle inter client
```
inter-client (or just "communication between clients")

rdb commented on programming/networking/distributed/ai-created-objects.rst:
```diff
@@ -0,0 +1,12 @@
+AI Side Created Objects
+=======================
+
+Similar to the client side created objects, AI server side created objects will
```
client-side, server-side

rdb commented on programming/networking/distributed/ai-created-objects.rst:
```diff
@@ -0,0 +1,12 @@
+=======================
+
+Similar to the client side created objects, AI server side created objects will
+be created and managed on the AI server. The difference here is, that the DOs
```
This comma should be removed

rdb commented on programming/networking/distributed/ai-repositories.rst:
```diff
@@ -0,0 +1,41 @@
+.. _ai-repositories:
+
+AI Server
+=========
```
Let's leave the title as "AI Repositories" since that's also what the ref says.

rdb commented on programming/networking/distributed/ai-repositories.rst:
```diff
@@ -0,0 +1,41 @@
+AI Server
+=========
+
+AI Servers are to be specific, clients that reside on a server rather then on
```
"to be specific" is extraneous here (though if you really wanted to put it here, I would say "To be specific, AI servers are…")

rdb commented on programming/networking/distributed/ai-repositories.rst:
```diff
@@ -0,0 +1,41 @@
+side. Clients shouldn’t be trusted as it’s not possible to ensure that they
+haven’t been compromised in one way or another.
+
+Similar to the Server Repositories, for AI Repositories most of the low-level
```
"repositories" should be lowercased in both cases here, as in both cases it is not a class name, proper noun or title.

rdb commented on programming/networking/distributed/ai-repositories.rst:
```diff
@@ -0,0 +1,41 @@
+       threadedNet = True)
+
+The setup is quite similar to the one of a normal client repository which we
+will take a look at in the next sections. The main difference is, that for an AI
```
Misplaced comma (just remove it)

rdb commented on programming/networking/distributed/ai-repositories.rst:
```diff
@@ -0,0 +1,41 @@
+   def deallocateChannel(self, doID):
+       print("Client left us: ", doID)
+
+This function will be called, whenever a client has disconnected and gives us the chance to react to it’s disconnection.
```
Remove comma

Also "it's" should be "its" here (remember that "it's" is a contraction of "it is", hence the apostrophe).

rdb commented on programming/networking/distributed/client-created-objects.rst:
```diff
@@ -0,0 +1,40 @@
+       className = 'DGameObject',
+       zoneId = 2)
+
+As you have seen we’ve set the zonId to 2. This tells the object to live in this
```
zonId→zoneId

rdb commented on programming/networking/distributed/client-repositories.rst:
```diff
@@ -0,0 +1,149 @@
+
+Similar to the server repositories, client repositories are there to handle the
+lower level connection code to the server. As seen earlier in the AI
+Repositories chapter, client repositories doesn’t necessarily have to be on end
```
This should be a link if it's referring back to the AI Repositories chapter.

rdb commented on programming/networking/distributed/client-repositories.rst:
```diff
@@ -0,0 +1,149 @@
+lower level connection code to the server. As seen earlier in the AI
+Repositories chapter, client repositories doesn’t necessarily have to be on end
+user machines but can also be used on servers. A basic client repository code
+may look as follow or should at least implement the following set of functions
```
"as follows" or "as the following"

"code" is uncountable in this definition, so "a code" is invalid; I'd just say "A basic client repository implementation"

rdb commented on programming/networking/distributed/clients.rst:
```diff
@@ -0,0 +1,12 @@
+This is the part that the end users will see and have running on their own
+machines. These applications are primarily there to connect to the server,
+gather user input and handle the audiovisual output. They should not have handle
+much of the game logic as that’s handled by the AI Server.
```
I feel like there's a word missing here, perhaps "They should not have **to** handle"?

"AI Server" should be "AI server" here.

rdb commented on programming/networking/distributed/clients.rst:
```diff
@@ -0,0 +1,12 @@
+much of the game logic as that’s handled by the AI Server.
+
+The important parts which have to be implemented in client applications to work
+with the distributed network will be shown in the next chapters
```
Full stop missing at the end.

rdb commented on programming/networking/distributed/dc-definition-files.rst:
```diff
@@ -0,0 +1,240 @@
+
+Note if, for example, the name value gets changed after generation of the DO, it
+doesn't change the value automatically of the DO on the server. This has to be
+done manually. Tho, calling the distributed versions of the functions (as
```
"Tho" is not a word, maybe you meant "However", "Although" or "Though".

rdb commented on programming/networking/distributed/dc-definition-files.rst:
```diff
@@ -0,0 +1,240 @@
+
+Any python objects to be mapped for distributed networking should be imported
+here. A modified python syntax is used. In the first line DistributedObject.py
+and DistributedObjectAI.py will be mapped.
```
Python should be capitalised, it is a proper noun.

rdb commented on programming/networking/distributed/distributed-object.rst:
```diff
@@ -0,0 +1,71 @@
+.. _distributed-boject:
```
Typo

rdb commented on programming/networking/distributed/distributed-object.rst:
```diff
@@ -0,0 +1,71 @@
+===================
+
+Distributed Objects are the base for all things that should be managed over the
+distributed network. May it be players, level objects, the level itself, a
```
"This may include players, …"

rdb commented on programming/networking/distributed/distributed-object.rst:
```diff
@@ -0,0 +1,71 @@
+The Foo class will be used on clients and the FooAI class will be generated on
+the AI servers.
+
+An example Distributed Object class may look like the following:
```
If you write it like a class it should be as DistributedObject, otherwise just "distributed object"

rdb commented on programming/networking/distributed/distributed-object.rst:
```diff
@@ -0,0 +1,71 @@
+DistributedObjectAI respectively. Usually those classes will be filled with
+methods that follow the form of foo, d_foo and b_foo.
+
+The foo method is the one that will have an affect locally.
```
affect→effect

rdb commented on programming/networking/distributed/index.rst:
```diff
@@ -0,0 +1,47 @@
+**Note:** The following parts which are rarely used are not documented yet.
+
+* Uber Dog servers and objects
+* Owner View objects
```
It would be nicer to use proper Sphinx `.. note` directive for such notes.

rdb reviewed (changes requested):
Thanks a lot!  This is a great start.  I've added some nitpicks regarding spelling and grammar.

fireclawthefox committed:
Fixes and enhancements

Fixed typos/spelling/grammar
Use :meth: and :class: where appropriate
Use sphinx note notation where appropriate
Enhanced Distributed Object Documentation
Enhanced Distributed Class Definition Documentation

fireclawthefox committed:
Enhanced DO documentation

Better information about DO usage on AIs
Documented special methods in DOs
Minor fix in the DC documentation

rdb commented:
Merged, thanks so much!
